app/src/main/java/com/example/lapnguyen/weather_3/screen/mainscreen/MainPresenter.java

```diff
@@ -50,5 +73,43 @@ public void accept(@NonNull Throwable throwable) throws Exception {
                 });
         mCompositeDisposable.add(disposable);
     }
+
+    public void startGoogleApi() {
+        mGoogleApiClient.connect();
+    }
+
+    public void stopGoogleApi() {
+        mGoogleApiClient.disconnect();
+    }
+
+    @SuppressWarnings("MissingPermission")
+    public void onConnected(@Nullable Bundle bundle) {
+        mLocationRequest = LocationRequest.create();
+        mLocationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+        mLocationRequest.setInterval(1000);
+        LocationServices.FusedLocationApi.requestLocationUpdates(mGoogleApiClient, mLocationRequest,
+                this);
+    }
+
+    @Override
+    public void onConnectionSuspended(int i) {
+
+    }
+
+    @Override
+    public void onConnectionFailed(
+            @android.support.annotation.NonNull ConnectionResult connectionResult) {
+
+    }
+
+    @Override
+    public void onLocationChanged(Location location) {
+        System.out.println(location.toString());
```

Comment:
ERROR: [SystemPrintln] System.out.println is used
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#SystemPrintln